contracts/adapters/CharmAdapter.sol

```diff
@@ -0,0 +1,388 @@
+     * @notice Check if an options contract exist based on the passed parameters.
+     * @param optionTerms is the terms of the option contract
+     */
+    function optionsExist(ProtocolAdapterTypes.OptionTerms calldata optionTerms)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return lookupOToken(optionTerms) != address(0);
+    }
+
+    /**
+     * @notice Get the options contract's address based on the passed parameters
+     * @param optionTerms is the terms of the option contract
+     */
+    function getOptionsAddress(
+        ProtocolAdapterTypes.OptionTerms calldata optionTerms
+    ) external view override returns (address) {
+        return lookupOToken(optionTerms);
+    }
+
+    /**
+     * @notice Gets the premium to buy `purchaseAmount` of the option contract in ETH terms.
+     */
+    function premium(
+      ProtocolAdapterTypes.OptionTerms calldata optionTerms,
+      uint256 purchaseAmount
+    )
+        external
+        view
+        override
+        returns (uint256 cost)
+    {
+        address tokenAddress = lookupOToken(optionTerms);
+
+        require(tokenAddress != address(0), "Must be valid option terms!");
+
+        // Get strike index, whether is long
+        OptionType memory optionType = addressToOptionType[tokenAddress];
+
+        //get token
+        IOptionToken token = IOptionToken(tokenAddress);
+        //get market
+        IOptionMarket market = IOptionMarket(token.market());
+
+        cost = optionViews.getBuyOptionCost(market, optionType.isLongToken, optionType.strikeIndex, purchaseAmount);
+    }
+
+    /**
+     * @notice Amount of profit made from exercising an option contract (current price - strike price). 0 if exercising out-the-money.
+     * @param options is the address of the options contract
+     * @param amount is the amount of tokens or options contract to exercise. Only relevant for fungle protocols like Opyn
+     */
+    function exerciseProfit(
+        address options,
+        uint256,
+        uint256 amount
+    ) public view override returns (uint256 profit) {
+      //get token
+      IOptionToken token = IOptionToken(options);
+      //get market
+      IOptionMarket market = IOptionMarket(token.market());
+
+      // Get strike index, whether is long
+      OptionType memory optionType = addressToOptionType[options];
+
+      //profit of exercising
+      profit = optionViews.getSellOptionCost(market, optionType.isLongToken, optionType.strikeIndex, amount);
+    }
+
+    function canExercise(
+        address options,
+        uint256 optionID,
+        uint256 amount
+    ) public view override returns (bool) {
+        //get token
+        IOptionToken token = IOptionToken(options);
+        //get market
+        IOptionMarket market = IOptionMarket(token.market());
+
+        if (block.timestamp < market.expiryTime()) {
+            return false;
+        }
+        if (exerciseProfit(options, optionID, amount) > 0) {
```

Comment:
Since the `optionID` param doesn't do anything here, we can just use 0

```solidity
exerciseProfit(options, 0, amount)
```

Then you can comment out the `optionID` parameter from the `canExercise` function